File: src/main/java/no/nav/domain/pensjon/regler/beregning/penobjekter/KrigOgGammelYrkesskade.java
package no.nav.domain.pensjon.regler.beregning.penobjekter;

import no.nav.domain.pensjon.regler.beregning.Ytelseskomponent;
import no.nav.domain.pensjon.regler.kode.YtelsekomponentTypeCti;

import java.io.Serializable;

/**
 * Brukes i PREG kun ved g-omregning
 */
public class KrigOgGammelYrkesskade extends Ytelseskomponent implements Serializable {

    /**
     * Pensjonsgraden
     */
    private int pensjonsgrad;
    /**
     * grunnlag for utbetaling;
     */
    private int grunnlagForUtbetaling;
    /**
     * Kapital utløsning
     */
    private int kapitalutlosning;
    /**
     * poengtall
     */
    private double ps;
    /**
     * forholdstall yg
     */
    private double yg;
    /**
     * Men del
     */
    private int mendel;

    public KrigOgGammelYrkesskade() {
        super();
        ytelsekomponentType = new YtelsekomponentTypeCti("KRIG_GY");
    }
}

File: src/main/kotlin/no/nav/pensjon/regler/domain/beregning/penobjekter/KrigOgGammelYrkesskade.kt
package no.nav.pensjon.regler.domain.beregning.penobjekter

import no.nav.pensjon.regler.domain.beregning.Ytelseskomponent
import no.nav.pensjon.regler.domain.enum.YtelseskomponentTypeEnum
import java.io.Serializable

/**
 * Brukes i pensjon-regler kun ved g-omregning
 */
class KrigOgGammelYrkesskade : Ytelseskomponent(), Serializable {
    /**
     * Pensjonsgraden
     */
    var pensjonsgrad = 0

    /**
     * grunnlag for utbetaling;
     */
    var grunnlagForUtbetaling = 0

    /**
     * Kapital utløsning
     */
    var kapitalutlosning = 0

    /**
     * poengtall
     */
    var ps = 0.0

    /**
     * forholdstall yg
     */
    var yg = 0.0

    /**
     * Men del
     */
    var mendel = 0

    override val ytelsekomponentTypeEnum: YtelseskomponentTypeEnum = YtelseskomponentTypeEnum.KRIG_GY
}